frontend/canonical.php

```diff
@@ -124,6 +126,9 @@ public function check_canonical_url( $requested_url = '', $do_redirect = true )
 		}
 
 		if ( empty( $language ) ) {
+			if ( empty( $this->curlang ) ) {
+				return $requested_url;
+			}
```

Comment:
Why?

Reply:
I haven't a use case in mind where `$this->curlang` could be `null`. However if there is such case, these lines changes the behavior. Today we can reach the filter (with `null` as language). This wouldn't be possible anymore.
I don't see anything wrong with the current behavior (do you?) so I'd prefer to keep it to avoid any potential issue in case someone is exploiting the filter in this case.

Reply:
I added a `@var` comment in [f96b947](https://github.com/polylang/polylang/pull/1677/commits/f96b9478b0d6bb56687c3614e22f04f1d4b81bf5). The other solution would be to document the next filter with `@param PLL_Language|null $language The language detected.`.